include/libtorrent/aux_/session_udp_sockets.hpp

```diff
@@ -56,6 +59,42 @@ namespace libtorrent { namespace aux {
 		bool write_blocked = false;
 	};
 
+	struct outgoing_udp_socket final : session_udp_socket
+	{
+		explicit outgoing_udp_socket(io_service& ios, std::string const& dev, bool ssl_)
+			: session_udp_socket(ios), device(dev), ssl(ssl_) {}
+
+		// the name of the device the socket is bound to, may be empty
+		// if the socket is not bound to a device
+		std::string device;
+
+		// set to true if this is an SSL socket
+		bool ssl = false;
```

Comment:
are these two members meant to change during the lifetime of the socket, or can the be ``const``?